### PR TITLE
Feature/company settings/approval process

### PR DESCRIPTION
**Pull request by Nenad85**
Add company-settings/approval-process feature

### CODE REVIEW

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      .eq(0)
+      .type(aps.approvalProcessFor);
+    cy.get('mat-checkbox .mat-checkbox-inner-container').eq(0).click({ force: true });
+    cy.get('dib-approval-process-dialog dib-input input[placeholder="Search people/groups"]').eq(2).type(aps.approver);
+    cy.get('mat-checkbox input[type=checkbox]').eq(2).click({ force: true });
+    cy.get('dib-approval-process-dialog ui-button[type=success]').click();
+    cy.get('dib-approval-process-item .item__content').should('contain', aps.approver);
+  });
+
+  it('Deletes approval process item', () => {
```
Comment (sjova) [inline cypress/integration/dib-travel/company-settings/approval-process.spec.ts]: We should be consistent with the first lowercase letter.

File: cypress/fixtures/company-settings/approval-process.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "approvalProcessFor": "qa",
+  "approver": "Group"
```
Comment (sjova) [inline cypress/fixtures/company-settings/approval-process.json]: Maybe `approvedBy` is a better naming. What do you think?

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { ApprovalProcess } from '../../../models';
+
+describe('Approval Process Suite', () => {
+  let aps: ApprovalProcess;
```
Comment (sjova) [inline cypress/integration/dib-travel/company-settings/approval-process.spec.ts]: `aps` - Please do not use abbreviations in your code

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  before(() => {
+    cy.fixture('company-settings/approval-process').then((approvalProcessFormData) => {
+      aps = approvalProcessFormData;
+    });
+  });
+
+  beforeEach(() => {
+    cy.login();
```
Comment (sjova) [inline cypress/integration/dib-travel/company-settings/approval-process.spec.ts:56]: Your general use `beforeEach` should be something like:
```
cy.login();
cy.visit('/company-management/approval-process');
```

And testing only sidebar link (and confirming loaded content) should be a separate single test.

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get('dib-layout dib-hamburger-icon').click();
```
Comment (sjova) [inline cypress/integration/dib-travel/company-settings/approval-process.spec.ts]: Better selector: `dib-navbar dib-hamburger-icon`

File: cypress/models/company-settings/approval-process.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export interface ApprovalProcess {
+  approvalProcessFor: string;
+  approver: string;
```
Comment (sjova) [inline cypress/models/company-settings/approval-process.ts]: Same as above

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get('[href="/company-management/approval-process"]').click();
+  });
+
+  it('submits empty approval process form', () => {
+    cy.get('dib-approval-process ui-button[type=primary]').click();
+    cy.get('dib-approval-process-dialog ui-button[type=success]').click({ force: true });
```
Comment (sjova) [inline cypress/integration/dib-travel/company-settings/approval-process.spec.ts]: missing parent selector: `.cdk-overlay-container`

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get('dib-dialog-wrapper dib-approval-process-dialog').should('be.visible');
```
Comment (sjova) [inline cypress/integration/dib-travel/company-settings/approval-process.spec.ts]: same as above

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get('dib-approval-process .header__details__helptext').click();
+    cy.get('.cdk-overlay-container .modal-content').scrollTo('bottom');
+    cy.get('dib-dialog-wrapper dib-approval-process-helptext-dialog').should('be.visible');
+    cy.get('dib-dialog-wrapper .material-icons').click();
+  });
+
+  it('creates a new approval process', () => {
+    cy.get('dib-approval-process ui-button[type=primary]').click();
+    cy.get('dib-approval-process-dialog dib-input input[placeholder="Search people/groups"]')
+      .eq(0)
```
Comment (sjova) [inline cypress/integration/dib-travel/company-settings/approval-process.spec.ts]: Please try to create a selector without `eq(0)`. A better solution should be applied in all places.

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+      .type(aps.approvalProcessFor);
+    cy.get('mat-checkbox .mat-checkbox-inner-container').eq(0).click({ force: true });
```
Comment (sjova) [inline cypress/integration/dib-travel/company-settings/approval-process.spec.ts]: Are you sure that you should use the click on checkbox element?

**Review by sjova — COMMENTED** [review]
You're missing parent feature selector: `dib-company-management dib-approval-process ...`

And please revisit all selectors.

File: cypress/models/index.ts [hunk 11]
```diff
@@ -1,2 +1,3 @@
 export * from './dib-travel-accounts';
```
Comment (sjova) [inline cypress/models/index.ts]: Double-check A-Z order of exports

**Comment by sjova** [comment]
Please update Describe label as we documented in `ISSUE-TEMPLATE.md`.

File: cypress/fixtures/company-settings/approval-process.json (hunk 2 above)
Comment (sjova) [reply cypress/fixtures/company-settings/approval-process.json]: Maybe you can use this file name: `approval-form.json`

And later below usage like this (if I understand usage correctly):

```
approvalForm.peopleOrGroup
approvalForm.approvedBy
```

Comment (Nenad85) [reply cypress/fixtures/company-settings/approval-process.json]: I agree with that.

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts (hunk 10 above)
Comment (Nenad85) [reply cypress/integration/dib-travel/company-settings/approval-process.spec.ts]: You're right, better with check.

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get('mat-checkbox input[type=checkbox]').eq(2).click({ force: true });
+    cy.get('dib-approval-process-dialog ui-button[type=success]').click();
+    cy.get('dib-approval-process-item .item__content').should('contain', aps.approver);
+  });
+
+  it('Deletes approval process item', () => {
+    cy.get('dib-approval-process dib-approval-process-item ui-button[type=warning]').eq(0).click();
+    cy.get('mat-dialog-container confirmation-dialog ui-button[cancel=true').click();
+    cy.get('dib-approval-process dib-approval-process-item ui-button[type=warning]').eq(0).click();
+    cy.get('mat-dialog-container confirmation-dialog ui-button[type=warning').click();
```
Comment (sjova) [inline cypress/integration/dib-travel/company-settings/approval-process.spec.ts]: Please try to find selector without Angular Material classes (`mat-*`)

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get('dib-approval-process ui-button[type=primary]').click();
+    cy.get('dib-approval-process-dialog ui-button[type=success]').click({ force: true });
+    cy.get('dib-dialog-wrapper dib-approval-process-dialog').should('be.visible');
+  });
+
+  it('displays info about approval process', () => {
+    cy.get('dib-approval-process .header__details__helptext').click();
+    cy.get('.cdk-overlay-container .modal-content').scrollTo('bottom');
+    cy.get('dib-dialog-wrapper dib-approval-process-helptext-dialog').should('be.visible');
+    cy.get('dib-dialog-wrapper .material-icons').click();
```
Comment (sjova) [inline cypress/integration/dib-travel/company-settings/approval-process.spec.ts]: Try to find a better selector instead `.material-icons`

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+
+  it('creates a new approval process (all trips)', () => {
+    cy.get('dib-company-management dib-approval-process ui-button[type=primary]').click();
+    cy.get('.cdk-overlay-container dib-approval-process-dialog dib-input input')
+      .first()
+      .type(approvalForm.approvalProcessFor);
+    cy.get('.cdk-overlay-container dib-approval-process-dialog .members .user')
+      .contains(approvalForm.approvalProcessForContent)
+      .click();
+    cy.get('.cdk-overlay-container dib-approval-process-dialog #mat-radio-3').click();
```
Comment (sjova) [inline cypress/integration/dib-travel/company-settings/approval-process.spec.ts]: Replace `#mat-radio-3` with better strategy

File: cypress/integration/dib-travel/company-settings/approval-process.spec.ts [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+
+  it('creates a new approval process (only out of policy trips)', () => {
+    cy.get('dib-company-management dib-approval-process ui-button[type=primary]').click();
+    cy.get('.cdk-overlay-container dib-approval-process-dialog dib-input input')
+      .first()
+      .type(approvalForm.approvalProcessFor);
+    cy.get('.cdk-overlay-container dib-approval-process-dialog .members .user')
+      .contains(approvalForm.approvalProcessForContent)
+      .click();
+    cy.get('.cdk-overlay-container dib-approval-process-dialog #mat-radio-2').click();
```
Comment (sjova) [inline cypress/integration/dib-travel/company-settings/approval-process.spec.ts]: Same as above

**Pull request merged**